rdrf/report/reports/generator.py

```diff
@@ -197,141 +153,218 @@ def export_to_json(self, request):
-        if 'cde.value' not in df.columns and 'cde.values' not in df.columns:
-            df.drop(columns=['cfg', 'form', 'section', 'cde'], inplace=True)
-            return stream_to_csv(df)
-
-        # Merge the value and values columns together so we can pivot it
-        if 'cde.value' not in df.columns:
-            df['cde.value'] = None
-        if 'cde.values' in df.columns:
-            df['cde.value'] = df['cde.value'].combine_first(df['cde.values'])
-
-        # Capture the demographic cols required for the pivot index before we go and rename columns
-        demographic_cols = [col for col in df.columns if col == 'id' or col not in clinical_cols]
-
-        # Rename the columns being pivoted to occur alphabetically in the order desired in the output report
-        # https://github.com/pandas-dev/pandas/issues/17041#issuecomment-317576297
-        df.rename(inplace=True, columns={'cfg.defaultName': 'a.cfg.defaultName',
-                                         'cde.value': 'b.cde.value'})
-
-        # Pivot the cde values by their uniquely identifying columns (context form group, form, section, cde)
-        cfg_name_col, form_name_col, section_name_col, cde_name_col = get_cde_pivot_columns(self.report_design.cde_heading_format)
-
-        cde_pivot_cols = [cfg_name_col, 'cfg.sortOrder', 'cfg.entryNum', form_name_col, section_name_col, 'section.entryNum', cde_name_col]
-
-        pivoted = df.pivot(index=demographic_cols,
-                           columns=cde_pivot_cols,
-                           values=['a.cfg.defaultName', 'b.cde.value'])
-
-        # Re-order the columns
-        pivoted = pivoted.sort_index(axis=1, level=cde_pivot_cols)
-
-        # Remove context form group's sort order as we don't need to see this in the results.
-        pivoted = pivoted.droplevel('cfg.sortOrder', axis=1)
-
-        # Flatten the column levels into one, and reformat labels
-        pivoted.columns = pivoted.columns.to_series().str.join('_')
-        pivoted.columns = pivoted.columns.to_series().apply(self.__reformat_pivoted_column_labels)
-
-        # Remove null columns (caused by patients with no matching clinical data)
-        pivoted = pivoted.loc[:, pivoted.columns.notnull()]
-        # Remove duplicate columns (caused by repeated context form group's defaultName from how it's pivoted)
-        pivoted = pivoted.loc[:, ~pivoted.columns.duplicated()]
-
-        return stream_to_csv(pivoted)
+        def get_demographic_headers():
+            def get_flat_json_path(report_model, report_field, i=None):
+                if not report_field:
+                    return None
+                if report_model == 'patient':
+                    prefix = ''
+                else:
+                    prefix = f'{report_model}_'
+
+                field_json_path = re.sub(r"\s", "", report_field)
+                field_json_path = re.sub(r"(.+)({(.*)})", r"\1_\3", field_json_path)
```

Comment:
Needs explanation

Reply:
Fixed. Hopefully I haven't over-explained it now.